### PR TITLE
fix: add request validation for creating groups

### PR DESCRIPTION
**Pull request by Dotunj**
This adds additional request validation to the endpoint for creating a group. Addresses #290

### CODE REVIEW

**Review by jirevwe — APPROVED** [review]

File: server/group.go [hunk 1]
```diff
@@ -58,6 +61,41 @@ func (a *applicationHandler) CreateGroup(w http.ResponseWriter, r *http.Request)
...
+
+	if util.IsStringEmpty(string(newGroup.Config.Signature.Header)) {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid signature header", http.StatusBadRequest))
+		return
+	}
+
+	if util.IsStringEmpty(string(newGroup.Config.Signature.Hash)) {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid hash", http.StatusBadRequest))
+		return
+	}
```
Comment (danvixent) [inline server/group.go:92]: is this necessary since we're still going to check algo.M for it?

File: server/group.go [hunk 2]
```diff
@@ -58,6 +61,41 @@ func (a *applicationHandler) CreateGroup(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	if util.IsStringEmpty(string(newGroup.Config.Strategy.Type)) {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid strategy type", http.StatusBadRequest))
+		return
+	}
```
Comment (danvixent) [inline server/group.go:67]: is this necessary since we're still going to compare to config.DefaultStrategyProvider?

File: server/group.go (hunk 1 above)
Comment (Dotunj) [reply server/group.go:92]: I think this improves the experience of the API since we'll be returning the correct validation message.

Comment (danvixent) [reply server/group.go:92]: Yeah, true.

**Review by danvixent — APPROVED** [review]

**Pull request closed without merging**